utils/adapters/firebase.ts

```diff
@@ -0,0 +1,65 @@
+import { App as FirebaseApp } from "firebase-admin/app";
+import { getAuth } from "firebase-admin/auth";
+import { getFirestore } from "firebase-admin/firestore";
+import { randomBytes, secretbox } from "tweetnacl";
+import SolanaAuth, { Adapter } from "../SolanaAuth";
+
+/** FROM: https://github.com/dchest/tweetnacl-js/wiki/Examples */
+const newNonce = () => randomBytes(secretbox.nonceLength);
+
+/**
+ * A SolanaAuth Adapter to connect with Firebase using Firestore
+ * @param {import("firebase-admin/app").App} firebase - A firebase app
+ * @returns {Adapter} Adapter
+ */
+export const FirebaseAdapter = (firebase: FirebaseApp): Adapter => {
+  const auth = getAuth(firebase);
+  const db = getFirestore(firebase);
+
+  return {
+    checkNonce: async (pubkey: string) => {
+      const docRef = db.doc(`profiles/${pubkey}`);
+      const doc = await docRef.get();
+
+      if (doc.exists) {
+        return doc.data()?.nonce;
+      }
+      return undefined;
+    },
+
+    verifyTTL: async (pubkey: string) => {
+      const docRef = db.doc(`profiles/${pubkey}`);
+      const doc = await docRef.get();
+      const tll = doc.data()?.tll;
+      if (tll < +new Date()) {
+        return false;
+      }
+      return true;
+    },
+
+    createProfile: async (pubkey: string) => {
```

Comment:
Nonce and TTL creation is independent from firebase

WDYT of 

1. Renaming this function to something like

saveSigninAttempt or saveSigninChallenge or something

2. Passing nonce and TTL as method params